opwen_email_server/actions.py

```diff
@@ -238,6 +239,38 @@ def _new_email_id(cls, email: str) -> str:
         return sha256(email.encode('utf-8')).hexdigest()
 
 
+class ProcessServiceEmail(_Action):
+    def __init__(self,
+                 raw_email_storage: AzureTextStorage,
+                 email_storage: AzureObjectStorage,
+                 next_task: Callable[[str], None],
+                 email_parser: Callable[[str], dict] = None,
+                 email_formatter: Callable[[str], dict] = None):
```

Comment:
I believe this type signature is incorrect since `EmailFormatter` below returns `Optional[dict]`.

```suggestion
                 email_formatter: Callable[[str], Optional[dict]] = None):
```